src/loaders/OmeZarrLoader.ts

```diff
@@ -172,50 +201,52 @@ async function pickLevelToLoad(
 }
 
 class OMEZarrLoader implements IVolumeLoader {
-  multiscalePath?: string;
-  hasT?: boolean;
-  hasC?: boolean;
+  multiscaleDims?: number[][];
+  metadata?: OMEZarrMetadata;
+  axesTCZYX?: [number, number, number, number, number];
+  maxExtent?: LoadSpecExtent;
 
   async loadDims(loadSpec: LoadSpec): Promise<VolumeDims[]> {
     const store = new HTTPStore(loadSpec.url);
 
-    const imagegroup = loadSpec.subpath;
-
-    const data = await openGroup(store, imagegroup, "r");
+    const data = await openGroup(store, loadSpec.subpath, "r");
 
     // get top-level metadata for this zarr image
     const allmetadata = await data.attrs.asObject();
     // each entry of multiscales is a multiscale image.
     const imageIndex = imageIndexFromLoadSpec(loadSpec, allmetadata.multiscales);
-    const multiscales: OMEDataset[] = allmetadata.multiscales[imageIndex].datasets;
+    const datasets: OMEDataset[] = allmetadata.multiscales[imageIndex].datasets;
     const axes: Axis[] = allmetadata.multiscales[imageIndex].axes;
 
     const axisTCZYX = remapAxesToTCZYX(axes);
-    // ZYX
-    const spatialAxes = findSpatialAxesZYX(axisTCZYX);
 
     // Assume all axes have the same units - we have no means of storing per-axis unit symbols
-    const unitName = axes[spatialAxes[2]].unit;
-    const unitSymbol = unitNameToSymbol(unitName) || unitName || "";
+    const spaceUnitName = axes[axisTCZYX[4]].unit;
+    const spaceUnitSymbol = unitNameToSymbol(spaceUnitName) || spaceUnitName || "";
 
-    const dimsPromises = multiscales.map(async (multiscale): Promise<VolumeDims> => {
-      const shape = await fetchShapeOfLevel(store, imagegroup, multiscale);
+    const timeUnitName = axisTCZYX[0] > -1 ? axes[axisTCZYX[0]].unit : undefined;
+    const timeUnitSymbol = unitNameToSymbol(timeUnitName) || timeUnitName || "";
+
+    const dimsPromises = datasets.map(async (dataset): Promise<VolumeDims> => {
+      const level = await openArray({ store, path: dataset.path, mode: "r" });
+      const shape = level.meta.shape;
       if (shape.length != axes.length) {
         console.log("ERROR: shape length " + shape.length + " does not match axes length " + axes.length);
       }
 
-      const scale5d = getScale(multiscale);
+      const scale5d = getScale(dataset);
 
       const d = new VolumeDims();
-      d.subpath = "";
-      d.shape = [1, 1, 1, 1, 1];
+      d.subpath = loadSpec.subpath;
```

Comment:
this is making me remember to note that loadSpec really needs to have a `multiresolutionLevel: number` field instead of subpath.  `subpath` should be specific to ome-zarr only and is something that is mapped from the multiresolution level.

Reply:
Worth another issue?

Reply:
Yeah that could be a separate ticket